src/RemoteRepoApi.ts

```diff
@@ -0,0 +1,179 @@
+}
+
+export class RemoteRepoApi implements AzureExtensionApiProvider, IGit, vscode.Disposable {
+    private static _handlePool: number = 0;
+    private _providers = new Map<number, IGit>();
+
+    public get repositories(): RemoteRepository[] {
+        const ret: RemoteRepository[] = [];
+
+        this._providers.forEach(({ repositories }) => {
+            if (repositories) {
+                ret.push(...repositories);
+            }
+        });
+
+        return ret;
+    }
+
+    public async openRepository(uri: vscode.Uri): Promise<RemoteRepository | null> {
+        return this.repositories.find(repo => uri === repo.rootUri) || null;
+    }
+
+    public get state(): APIState | undefined {
+        if (this._providers.size === 0) {
+            return undefined;
+        }
+
+        for (const [, { state }] of this._providers) {
+            if (state !== 'initialized') {
+                return 'uninitialized';
+            }
+        }
+
+        return 'initialized';
+    }
+
+    // Remote Repositories extension needs to have this API exported to be used,
+    // so we need to merge it with the Static Web App extension API. The easiest way to do this is to just implement getApi on RemoteRepoApi
+    public getApi = createApiProvider([<AzureExtensionApi>{
+        revealTreeItem,
+        apiVersion: '1.0.0'
+    }]).getApi;
+
+    private _onDidOpenRepository = new vscode.EventEmitter<RemoteRepository>();
+    readonly onDidOpenRepository: vscode.Event<RemoteRepository> = this._onDidOpenRepository.event;
+    private _onDidCloseRepository = new vscode.EventEmitter<RemoteRepository>();
+    readonly onDidCloseRepository: vscode.Event<RemoteRepository> = this._onDidCloseRepository.event;
+    private _onDidChangeState = new vscode.EventEmitter<APIState>();
+    readonly onDidChangeState: vscode.Event<APIState> = this._onDidChangeState.event;
+    private _onDidPublish = new vscode.EventEmitter<PublishEvent>();
+    readonly onDidPublish: vscode.Event<PublishEvent> = this._onDidPublish.event;
+
+    private _disposables: vscode.Disposable[];
+    constructor() {
+        this._disposables = [];
+    }
+
+    private _updateReposContext() {
+        const reposCount = Array.from(this._providers.values()).reduce((prev, current) => {
+            return prev + current.repositories.length;
+        }, 0);
+        void vscode.commands.executeCommand('setContext', 'gitHubOpenRepositoryCount', reposCount);
+    }
+
+    registerGitProvider(provider: IGit): vscode.Disposable {
+        const handle = this._nextHandle();
+        this._providers.set(handle, provider);
+
+        this._disposables.push(provider.onDidCloseRepository(e => this._onDidCloseRepository.fire(e)));
+        this._disposables.push(provider.onDidOpenRepository(e => {
+            this._updateReposContext();
+            this._onDidOpenRepository.fire(e);
+        }));
+        if (provider.onDidChangeState) {
+            console.log(provider.onDidChangeState);
```

Comment:
Test line?

Reply:
Oh yeah, that API is pulled from somewhere else. Will clean it up 👍

Reply:
Oh, this isn't our source? The original source should be noted in a comment

Reply:
Thanks for reminding me; I was meaning to ask.

The repo that this was pulled from is private, so I wasn't sure if I should list the original source. It's also not exactly the same since we didn't need everything that they implemented, but just needed a sample for the `registerGitProvider` function.